state/protocol/badger/snapshot_test.go

```diff
@@ -806,34 +818,37 @@ func TestLatestSealedResult(t *testing.T) {
 			err = state.ExtendCertified(context.Background(), block2, block3.Header.QuorumCertificate())
 			require.NoError(t, err)
 
-			// B1 <- B2(R1,S1)
-			// querying B2 should return result R1, seal S1
+			err = state.ExtendCertified(context.Background(), block3, block4.Header.QuorumCertificate())
+			require.NoError(t, err)
+
+			// B1 <- B2(R1) <- B3(S1)
+			// querying B3 should return result R1, seal S1
 			t.Run("reference block contains seal", func(t *testing.T) {
-				gotResult, gotSeal, err := state.AtBlockID(block2.ID()).SealedResult()
+				gotResult, gotSeal, err := state.AtBlockID(block3.ID()).SealedResult()
 				require.NoError(t, err)
 				assert.Equal(t, block2.Payload.Results[0], gotResult)
-				assert.Equal(t, block2.Payload.Seals[0], gotSeal)
+				assert.Equal(t, block3.Payload.Seals[0], gotSeal)
 			})
 
-			err = state.ExtendCertified(context.Background(), block3, block4.Header.QuorumCertificate())
+			err = state.ExtendCertified(context.Background(), block4, block5.Header.QuorumCertificate())
 			require.NoError(t, err)
 
-			// B1 <- B2(R1,S1) <- B3
+			// B1 <- B2(S1) <- B3(S1)
 			// querying B3 should still return (R1,S1) even though they are in parent block
```

Comment:
```suggestion
			// B1 <- B2(S1) <- B3(S1) <- B4(R2,R3)
			// querying B3 should still return (R1,S1) even though they are in parent block
```
I am confused about this wording: 
> querying **B3** should still return (R1,S1) even though they are in **parent** block

I think it makes sense to query **B4** (or both B3 and B4).